presto-product-tests-launcher/src/main/java/io/prestosql/tests/product/launcher/env/EnvironmentModule.java

```diff
@@ -14,26 +14,36 @@
 package io.prestosql.tests.product.launcher.env;
 
 import com.google.inject.Binder;
+import com.google.inject.Inject;
 import com.google.inject.Module;
+import com.google.inject.Provides;
+import com.google.inject.Singleton;
 import com.google.inject.multibindings.MapBinder;
+import io.prestosql.tests.product.launcher.PathResolver;
 import io.prestosql.tests.product.launcher.env.common.Hadoop;
 import io.prestosql.tests.product.launcher.env.common.Kafka;
 import io.prestosql.tests.product.launcher.env.common.Kerberos;
 import io.prestosql.tests.product.launcher.env.common.KerberosKms;
 import io.prestosql.tests.product.launcher.env.common.Standard;
 import io.prestosql.tests.product.launcher.testcontainers.PortBinder;
 
+import java.io.File;
+
 import static com.google.inject.multibindings.MapBinder.newMapBinder;
+import static io.prestosql.tests.product.launcher.env.Environments.nameForConfigClass;
 import static java.util.Objects.requireNonNull;
 
 public final class EnvironmentModule
         implements Module
 {
     public static final String BASE_PACKAGE = "io.prestosql.tests.product.launcher.env.environment";
+    public static final String BASE_CONFIG_PACKAGE = "io.prestosql.tests.product.launcher.env.configs";
+    private final EnvironmentOptions environmentOptions;
```

Comment:
call it `baseEnvironmentOptions` and instead of mutating it create new instance based on it in `provideEnvironmentOptions`